### PR TITLE
Keyword search feature

### PR DESCRIPTION
**Pull request by ericguzman**
demo: https://ericguzman.github.io/debugger-protocol-viewer/

Summary of search feature:
+ Introduces search index and search index creation script.
+ Adds search control (custom Polymer element) to page header that activates/deactivates the search field.
+ Adds search results menu (custom Polymer element) presenting matches.

image

### CODE REVIEW

**Comment by ericguzman** [comment]
@kdzwinel, @paulirish 
Thanks again for your suggestions. I think I've addressed them with this implementation, as well as added many UI improvements. Take a look and feel free to provide feedback:
http://ericguzman.github.io/debugger-protocol-viewer/

It's not perfect but it should be production-worthy (anyone have a good icon for the category of "Method"?). Please let me know if you spot any blockers.

**Comment by paulirish** [comment]
Whoa. This is nice stuff. I like the UI!

* The 200ms for the debounce feels a little long to me. How about something more like 70?

Outside of that, I'm pretty happy. @kdzwinel I'll let you have a crack at this PR before green-buttoning it.

File: _layouts/default.html [hunk 1]
```diff
@@ -31,14 +34,15 @@
 
   <paper-header-panel main mode="waterfall">
     <paper-toolbar id="mainToolbar">
-      <paper-icon-button id="paperToggle" icon="menu" paper-drawer-toggle></paper-icon-button>
-      <span class="flex"></span>
-
-      <div class="bottom title">Chrome Debugger Protocol Viewer</div>
+      <paper-icon-button id="paperToggle"
+          icon="menu" paper-drawer-toggle></paper-icon-button>
+      <!-- <span class="flex"></span> -->
```
Comment (kdzwinel) [inline _layouts/default.html]: :shower: lets remove that

File: _layouts/default.html [hunk 2]
```diff
@@ -50,7 +54,18 @@
...
+  var toolbar = document.getElementById('mainToolbar');
+  btn.addEventListener('active', function() {
+    toolbar.classList.add('search-active');
+  }, false);
+  btn.addEventListener('inactive', function() {
+    toolbar.classList.remove('search-active');
+  }, false);
+});
+</script>
+<!-- {  % include github-badge.html %  } -->
```
Comment (kdzwinel) [inline _layouts/default.html]: Oh, the Github badge was obscuring the search box, right? We still want people to know that they can contribute to that project though. Maybe we should just mention that on the main page ( https://ericguzman.github.io/debugger-protocol-viewer/ )? @paulirish WDYT?

**Comment by kdzwinel** [comment]
Looks amazing, I love the icons in the search results :heart: Also, huge props for looking into the performance issues!

Code review of 38k of lines is challenging, especially because Github won't show me the full diff, but I caught these:

1. I had a quick look at `searchindex.json`:

screen shot 2015-09-29 at 00 43 44

TBH I don't see why we have both `href` and `domainHref` there. It probably gzips well, so maybe it's a non-issue, but if it's simple to remove `domainHref` or reduce `href` just to a hash part, it'd be great.

2. I observed that whenever I start searching new font file is loaded by the paper-ripple element. It looks like a polymer quirk, but maybe you'll know how to fix that?

3. Since `searchindex.json` has to be rebuilt after each `protocol.json` lets make sure to mention that in the readme: https://github.com/ChromeDevTools/debugger-protocol-viewer#building

File: _layouts/default.html (hunk 2 above)
Comment (ericguzman) [inline _layouts/default.html]: Oops, I forgot about that change. Yeah, it was in the way, but I'm all for bringing it back (affixed to a different corner, of course). I'll wait for Paul to reply before adding it back.

Comment (paulirish) [inline _layouts/default.html]: Yup. lets just add it after an `<hr>` on the home page or whatever

**Comment by paulirish** [comment]
@ericguzman looks like your commit is to address our comments. should we take another look?

**Comment by ericguzman** [comment]
Had a couple more cleanup items + merged upstream changes. All done now, so
please take a look.

**Comment by ericguzman** [comment]
@kdzwinel - Just a few more notes on code updates per your review:

1. Took your suggestion regarding the search data optimization by adding link construction logic to the view. With a couple of other adjustments, I was able to bring the size down from ~150kb to ~115, which gzips to about 18kb.
2. Hmm. I can’t repro this issue.
3. Done.

**Comment by kdzwinel** [comment]
There are no updates to the `searchindex.json` after merging upstream changes? https://github.com/ericguzman/debugger-protocol-viewer/commit/a7b93faac7104f1215e1a1de34d6a53e6c9612b7

2. I'm able to reproduce it every time: https://dl.dropboxusercontent.com/u/8989748/font.mov

**Comment by kdzwinel** [comment]
@ericguzman FYI I'm absolutely OK with merging that PR after you updated the search index. Double font loading issue is small and can be resolved later on.

**Comment by paulirish** [comment]
did https://github.com/ericguzman/debugger-protocol-viewer/commit/41791f1d80c41fe305a725df4e4a42826a1a56e1 update the index?

**Comment by ericguzman** [comment]
@paulirish Yup. 

I've created an issue for fixing the double font load issue. I found that I couldn't repro on my machines, presumably because I have Roboto installed locally on them. Using a linux box at work showed the issue, and I have a couple things to try next. Thanks for the vid, @kdzwinel.

So we're good to go with this PR now?

**Pull request merged**